### ISSUE TITLE
unit testing without postgres dependency (mock)

### ISSUE THREAD

**Issue by avelino** [prest/prest#579]
### Discussed in https://github.com/prest/prest/discussions/562

<div type='discussions-op-text'>

<sup>Originally posted by **avelino** June 23, 2021</sup>
The `postgres` adapter depends on the postgreSQL (database) server to run (`go test`), i.e. it was developed with integration testing.

Integration testing is necessary, but it raises the barrier for new members to contribute, i.e. for a person to make any contribution he needs to learn to configure the development environment and only then contribute. I think it's great to have people immersed in the project, but this is not the reality of open source projects that people contribute in their spare time (usually without pay)

I don't think it is necessary to "remove" the integration test but put it in its place:
1. unit test that does not depend on external services (e.g. PostgreSQL) running with `go test ....`, so our people were able to run the tests locally (on their machines)
2. Integration test: run only on CI (**GitHub Actions** or Travis CI) on pull request - doesn't necessarily have to be written with `go test`

What would be the best option to write `postgres` adapter **mock**?

Ref:
- [ ] #510
- [ ] #555 
- [ ] #558 
- [ ] #472 
- … as we see above we have several issues related to this subject - sad</div>

**Comment by vmesel** [comment]
Is this still open?

**Comment by avelino** [comment]
> Is this still open?

@vmesel yep, we need to rewrite the tests with unit tests in mind, this simplifies the way to run the tests for the incoming contributors

Leave the integration test (e2e) as we do for the build process (in the pipeline, even so we need to think if it is necessary)
